docs/migration.md

```diff
@@ -0,0 +1,193 @@
+# Prometheus 2.0 Migration Guide.
+
+In line with our [compatibility promise](TODO), the Prometheus 2.0 release contains
```

Comment:
*compatibility

Reply:
Don't think we have a "compatibility promise" section anywhere in the docs, so how about just `In line with [semantic versioning](http://semver.org/), ...`?

Reply:
It's not in line with semver, as we have quite a few exclusions.